pkg/karmadactl/promote.go

```diff
@@ -172,81 +184,81 @@ func (o *CommandPromoteOption) Validate() error {
 		return fmt.Errorf("output format is only one of json and yaml")
 	}
 
-	// If '--cluster-context' not specified, take the cluster name as the context.
-	if len(o.ClusterContext) == 0 {
-		o.ClusterContext = o.Cluster
-	}
-
 	return nil
 }
 
-// RunPromote promote resources from legacy clusters
-func RunPromote(karmadaConfig KarmadaConfig, opts CommandPromoteOption, args []string) error {
+// Run promote resources from legacy clusters
+func (o *CommandPromoteOption) Run(f util.Factory, args []string) error {
 	// Get control plane karmada-apiserver client
-	controlPlaneRestConfig, err := karmadaConfig.GetRestConfig(opts.KarmadaContext, opts.KubeConfig)
+	karmadaClient, err := f.KarmadaClientSet()
 	if err != nil {
-		return fmt.Errorf("failed to get control plane rest config. context: %s, kubeconfig: %s, error: %v",
-			opts.KarmadaContext, opts.KubeConfig, err)
+		return fmt.Errorf("failed to get control plane client. error: %w", err)
 	}
 
-	clusterInfos := make(map[string]*ClusterInfo)
-
-	if err := getClusterInKarmada(controlPlaneRestConfig, clusterInfos); err != nil {
-		return fmt.Errorf("failed to get cluster info in karmada control plane. err: %v", err)
+	clusterInfos, err := GetClusterInKarmada(karmadaClient)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster info in karmada control plane. err: %w", err)
 	}
 
-	if _, exist := clusterInfos[opts.Cluster]; !exist {
-		return fmt.Errorf("cluster(%s) does't exist in karmada control plane", opts.Cluster)
+	if _, exist := clusterInfos[o.Cluster]; !exist {
+		return fmt.Errorf("cluster(%s) does't exist in karmada control plane", o.Cluster)
```

Comment:
the code logic of L198 - L204 is conflicted with the L209 - L219.

I think the  L198 - L204  is redundant.